WcaOnRails/spec/models/competition_spec.rb

```diff
@@ -2,11 +2,24 @@
 
 RSpec.describe Competition do
   it "defines a valid competition" do
-    competition = FactoryGirl.build :competition, name: "Foo !Test- 2015"
+    competition = FactoryGirl.build :competition, name: "Foo: Test - 2015"
     expect(competition).to be_valid
     expect(competition.id).to eq "FooTest2015"
-    expect(competition.name).to eq "Foo !Test- 2015"
-    expect(competition.cellName).to eq "Foo !Test- 2015"
+    expect(competition.name).to eq "Foo: Test - 2015"
+    expect(competition.cellName).to eq "Foo: Test - 2015"
+  end
+
+  it "rejects invalid names" do
+    competition = FactoryGirl.build :competition, name: "Foo (Test)- 2015"
+    expect(competition).to be_invalid
+    competition = FactoryGirl.build :competition, name: "Poly^3 2016"
+    expect(competition).to be_invalid
+    competition = FactoryGirl.build :competition, name: "HOOAH! SMA 2015"
+    expect(competition).to be_invalid
+    competition = FactoryGirl.build :competition, name: "Campeonato de Cubos Mágicos de São Carlos/SP 2013"
+    expect(competition).to be_invalid
+    competition = FactoryGirl.build :competition, name: "Moldavian Nationals – Winter 2016"
+    expect(competition).to be_invalid
```

Comment:
I think this can be really cleaned up by a loop
```ruby
%w(name1, name2, ...).each do |name|
  expect(FactoryGirl.build(:competition, name: name).to be_invalid
end
```